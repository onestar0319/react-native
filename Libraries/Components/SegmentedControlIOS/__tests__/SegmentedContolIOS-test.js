/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 */

'use strict';

const React = require('react');
const ReactTestRenderer = require('react-test-renderer');

const SegmentedControlIOS = require('../SegmentedControlIOS.ios');

describe('SegmentedControlIOS', () => {
  it('renders the segmented control', () => {
    const component = ReactTestRenderer.create(<SegmentedControlIOS />);
    expect(component).not.toBeNull();
  });
  it('renders the segmented control with enabled default value', () => {
    const component = ReactTestRenderer.create(<SegmentedControlIOS />);
    expect(component.toTree().rendered.props.enabled).toBe(true);
    expect(component).toMatchSnapshot();
  });
  it('renders the segmented control with enabled', () => {
    const component = ReactTestRenderer.create(
      <SegmentedControlIOS enabled={true} />,
    );
    expect(component.toTree().rendered.props.enabled).toBe(true);
    expect(component).toMatchSnapshot();
  });
  it('renders the segmented control with enabled set to false', () => {
    const component = ReactTestRenderer.create(
      <SegmentedControlIOS enabled={false} />,
    );
    expect(component.toTree().rendered.props.enabled).toBe(false);
    expect(component).toMatchSnapshot();
  });
  it('renders the segmented control with values default value', () => {
    const component = ReactTestRenderer.create(<SegmentedControlIOS />);
    expect(component.toTree().rendered.props.values).toEqual([]);
    expect(component).toMatchSnapshot();
  });
  it('renders the segmented control with values', () => {
    const values = ['One', 'Two'];
    const component = ReactTestRenderer.create(
      <SegmentedControlIOS values={values} />,
    );
    expect(component.toTree().rendered.props.values).toBe(values);
    expect(component).toMatchSnapshot();
  });
});
