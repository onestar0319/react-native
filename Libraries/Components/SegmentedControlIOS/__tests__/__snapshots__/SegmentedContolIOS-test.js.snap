// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControlIOS renders the segmented control with enabled 1`] = `
<RCTSegmentedControl
  enabled={true}
  onChange={[Function]}
  style={
    Array [
      Object {
        "height": 28,
      },
      undefined,
    ]
  }
  values={Array []}
/>
`;

exports[`SegmentedControlIOS renders the segmented control with enabled default value 1`] = `
<RCTSegmentedControl
  enabled={true}
  onChange={[Function]}
  style={
    Array [
      Object {
        "height": 28,
      },
      undefined,
    ]
  }
  values={Array []}
/>
`;

exports[`SegmentedControlIOS renders the segmented control with enabled set to false 1`] = `
<RCTSegmentedControl
  enabled={false}
  onChange={[Function]}
  style={
    Array [
      Object {
        "height": 28,
      },
      undefined,
    ]
  }
  values={Array []}
/>
`;

exports[`SegmentedControlIOS renders the segmented control with values 1`] = `
<RCTSegmentedControl
  enabled={true}
  onChange={[Function]}
  style={
    Array [
      Object {
        "height": 28,
      },
      undefined,
    ]
  }
  values={
    Array [
      "One",
      "Two",
    ]
  }
/>
`;

exports[`SegmentedControlIOS renders the segmented control with values default value 1`] = `
<RCTSegmentedControl
  enabled={true}
  onChange={[Function]}
  style={
    Array [
      Object {
        "height": 28,
      },
      undefined,
    ]
  }
  values={Array []}
/>
`;
